### ISSUE TITLE
NodeFeature:SetHostnameAsFQDN failing in periodic node-kubelet-features-master job

### ISSUE THREAD

**Issue by harche** [kubernetes/kubernetes#98020]
**Which jobs are failing**:
[sig-node-kubelet/node-kubelet-features-master ](https://testgrid.k8s.io/sig-node-kubelet#node-kubelet-features-master) 

**Which test(s) are failing**:
https://github.com/kubernetes/kubernetes/blob/23839ba6492a682d5011d112e3772d4f5978b2de/test/e2e_node/pod_hostnamefqdn_test.go

**Since when has it been failing**:
From about 9 hours ago.

**Testgrid link**:
https://testgrid.k8s.io/sig-node-kubelet#node-kubelet-features-master

**Reason for failure**:
Unknown. 

**Anything else we need to know**:
Along with this job, even [CRIO job](https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-features) for `NodeFeatures` started failing with similar tests. 

While `node-kubelet-features-master` is failing, [node-kubelet-features-1.20](https://testgrid.k8s.io/sig-node-kubelet#node-kubelet-features-1.20) is not failing indicating that something went in master in last 9 hours that's breaking the tests.

**Comment by harche** [comment]
One interesting node on CRIO job is, it's only failing when run by test-infra. If I trigger those `NodeFeatures` jobs locally it completes successfully. Maybe something broke in `test-infra`? not sure. 

```
export KUBE_SSH_USER=core && make test-e2e-node REMOTE=true RUNTIME=remote IMAGE_CONFIG_FILE="/home/harshal/Downloads/image-config-cgrpv1.yaml" INSTANCE_PREFIX="harpatil" CLEANUP=false CONTAINER_RUNTIME_ENDPOINT="unix:///var/run/crio/crio.sock" FOCUS="\[Feature:SetHostnameAsFQDN\]" SKIP="\[Flaky\]|\[Slow\]|\[Serial\]" TEST_ARGS='--kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/crio.service --non-masquerade-cidr=0.0.0.0/0" --extra-log="{\"name\": \"crio.log\", \"journalctl\": [\"-u\", \"crio\"]}"'
```
That's the way to trigger node e2e from your local system (make sure gcloud is configured because just like test-infra, the actual tests are run on gcloud)

**Comment by hasheddan** [comment]
I believe I have identified the issue stemming from https://github.com/kubernetes/kubernetes/pull/96315. Working on fix now :+1: 

/assign

**Comment by hasheddan** [comment]
Fix in #98025
